packages/@azure/servicebus/data-plane/test/stress/messageRandomDisposition.ts

```diff
@@ -0,0 +1,139 @@
+import { ServiceBusClient, SendableMessageInfo, OnMessage, OnError, delay } from "../../lib";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 5 * 12 * 24 * 7; // 1 week
+
+const messagesToProcess: Set<number> = new Set<number>();
+const messageAbandonedMap: { [key: number]: number } = {};
+
+let abandonCount = 0;
+let completeCount = 0;
+let deadletterCount = 0;
+let deferCount = 0;
+
+let msgId = 1;
+
+let snapshotIntervalID: any;
+
+let isJobDone = false;
+
+async function main(): Promise<void> {
+  snapshotIntervalID = setInterval(snapshot, 5000); // Every 5 seconds
+  sendMessages();
+  receiveMessages();
+}
+
+async function sendMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+  try {
+    const sender = client.createSender();
+
+    while (!isJobDone) {
+      const message: SendableMessageInfo = {
+        messageId: msgId,
+        body: "test",
+        label: `${msgId}`
+      };
+      messageAbandonedMap[msgId] = 0;
+      messagesToProcess.add(msgId);
+      msgId++;
+      await sender.send(message);
+      await delay(2000); // Throttling send to not increase queue size
+    }
+  } finally {
+    client.close();
+    ns.close();
+  }
+}
+
+async function receiveMessages(): Promise<void> {
+  const ns = ServiceBusClient.createFromConnectionString(connectionString);
+  const client = ns.createQueueClient(queueName);
+
+  try {
+    const receiver = client.createReceiver();
+    const onMessageHandler: OnMessage = async (brokeredMessage) => {
+      const receivedMsgId = brokeredMessage.messageId;
+
+      if (typeof receivedMsgId !== "number") {
+        throw new Error("MessageId is corrupt or is of unexpected type");
+      }
+
+      const seed = Math.floor((Math.random() * 10) % 4);
+
+      switch (seed) {
+        case 0: {
+          const currCount = messageAbandonedMap[receivedMsgId];
+          if (currCount === 10) {
+            abandonCount++;
+            if (messagesToProcess.has(receivedMsgId)) {
+              messagesToProcess.delete(receivedMsgId);
+            }
+          }
+          messageAbandonedMap[receivedMsgId] = currCount + 1;
+          brokeredMessage.abandon();
+          break;
+        }
+        case 1: {
+          completeCount++;
+          if (messagesToProcess.has(receivedMsgId)) {
+            messagesToProcess.delete(receivedMsgId);
+          }
+          brokeredMessage.complete();
+          break;
+        }
+        case 2: {
+          deadletterCount++;
+          if (messagesToProcess.has(receivedMsgId)) {
```

Comment:
Since these are long running tests, I wanted to be careful about not having tight checks and allowing the tests to run inspite of few anomalies (and not necessarily crash, stop the test).

In this case, we could log the error versus throwing just to keep the tests running?
Or, should this be updated to have a stricter verification?

Reply:
Fair -  we can log it -- but I am fine if we don't do anything about it right now and leave the code as-is.